src/pages/endpoint/index.js

```diff
@@ -0,0 +1,200 @@
+import * as React from 'react'
+import styled from 'styled-components'
+import { Formik, Form } from 'formik'
+import { WithIntl } from 'components/localization'
+import Layout from 'components/layout/layout'
+import { Container, SEO } from 'components/containers'
+import { Header, Text } from 'components/elements'
+import { Input, Button } from 'components/form'
+import validation from 'common/validation'
+import { trimSpaces } from 'common/utility'
+import { default_server_url } from 'common/constants'
+import { getAppId } from 'common/websocket/config'
+import { DerivStore } from 'store'
+import { useLocalStorageState } from 'components/hooks/use-localstorage-state'
+
+const StyledContainer = styled(Container)`
+    text-align: center;
+    height: 100vh;
+    padding: auto 0;
+    justify-content: start;
+`
+
+const ButtonContainer = styled.div`
+    margin-top: 2rem;
+`
+
+const InputGroup = styled.div`
+    width: 40rem;
+    margin: 0 auto 3.4rem;
+`
+
+const StyledButton = styled(Button)`
+    margin: 0.8rem 0.4rem;
+`
+
+const endpointValidation = (values) => {
+    let errors = {}
+
+    const server_url = trimSpaces(values ? values.server_url : '')
+    const app_id = trimSpaces(values ? values.app_id.toString() : '')
+    const server_url_error = validation.text(server_url)
+    const app_id_error = validation.text(app_id)
+
+    if (server_url_error) {
+        errors.server_url = server_url_error
+    }
+
+    if (app_id_error) {
+        errors.app_id = app_id_error
+    }
+
+    return errors
+}
+
+const Endpoint = () => {
+    const [server_url, setServerUrl] = useLocalStorageState(default_server_url, 'config.server_url')
+    const [app_id, setAppId] = useLocalStorageState(getAppId(), 'config.app_id')
```

Comment:
This, 'config.app_id' looks like a key inside an object. Should this be in string? Or can we rename the argument?

Reply:
its not an object, its a name convention used by backend and qa to set app_id and server_url, changing this will also require them to change their automation test